src/gnssro/gnss_tec_bufr2ioda.py

```diff
@@ -0,0 +1,239 @@
+    VarAttrs[('SATxECFPosition', 'MetaData')]['units'] = 'km'
+    VarAttrs[('SATyECFPosition', 'MetaData')]['units'] = 'km'
+    VarAttrs[('SATzECFPosition', 'MetaData')]['units'] = 'km'
+    VarAttrs[('dateTime', 'MetaData')]['units'] = iso8601_string
+
+    VarAttrs[('totalElectronContent', 'ObsValue')]['_FillValue'] = float_missing_value
+    VarAttrs[('totalElectronContent', 'ObsError')]['_FillValue'] = float_missing_value
+    VarAttrs[('totalElectronContent', 'PreQC')]['_FillValue'] = int_missing_value
+
+    VarAttrs[('elevationAngleGNSSlink', 'MetaData')]['_FillValue'] = float_missing_value
+    VarAttrs[('GNSSxECFPosition', 'MetaData')]['_FillValue'] = float_missing_value 
+    VarAttrs[('GNSSyECFPosition', 'MetaData')]['_FillValue'] = float_missing_value
+    VarAttrs[('GNSSzECFPosition', 'MetaData')]['_FillValue'] = float_missing_value
+    VarAttrs[('SATxECFPosition', 'MetaData')]['_FillValue'] = float_missing_value
+    VarAttrs[('SATyECFPosition', 'MetaData')]['_FillValue'] = float_missing_value
+    VarAttrs[('SATzECFPosition', 'MetaData')]['_FillValue'] = float_missing_value
+
+    # final write to IODA file
+    writer.BuildIoda(obs_data, VarDims, VarAttrs, GlobalAttrs)
+
+
+def get_meta_data(ds):
+
+    # get some of the global attributes that we are interested in
+    meta_data_keys = def_meta_data()
+
+    # these are the MetaData we are interested in
+    profile_meta_data = {}
+    psize = len(ds['time'])
+    for k, v in meta_data_keys.items():
+        try:
+            profile_meta_data[k] = getattr(ds, v)
+        except Exception as e:
+            print(f'  WARNING: could not retrieve key: {k} -- Skipping')
+        if type(v) is np.int64:
+            obs_data[(k, 'MetaData')] = np.array(np.repeat(v, psize), dtype=np.int64)
+        elif type(v) is int:
+            obs_data[(k, 'MetaData')] = np.array(np.repeat(v, psize), dtype=ioda_int_type)
+        elif type(v) is float:
+            obs_data[(k, 'MetaData')] = np.array(np.repeat(v, psize), dtype=ioda_float_type)
+        else:  # something else (what do we do with it)
+            print(f"Found neither float nor in, type={type(v)}; skipping")
+
+    time_offset = np.array(ds['time'].add_offset - ds['time'][:])
+    profile_meta_data['dateTime'] = np.int64(time_offset)
+
+    return profile_meta_data
+
+
+def def_meta_data():
+
+    meta_data_keys = {
+#       "elevationAngleGNSSlink": 'elevation',
+#       "GNSSxECFPosition": 'x_GPS',
+#       "GNSSyECFPosition": 'y_GPS',
+#       "GNSSzECFPosition": 'z_GPS',
+#       "SATxECFPosition": 'x_GPS',
+#       "SATyECFPosition": 'y_GPS',
+#       "SATzECFPosition": 'z_GPS',
+        "satelliteTransmitterId": 'prn_id',
+#       "satelliteConstellationRO": 'satelliteClassification',
+    }
+
+    return meta_data_keys
+
+
+def def_meta_types():
+
+    meta_data_types = {
+        "dateTime": "long",
+        "elevationAngleGNSSlink": "float",
+        "GNSSxECFPosition": "float",
```

Comment:
@davidkuhl I'm coming up with these new IODA entry names, but we'll need to but a bit more formal and likely should ask NOAA SWPC but trying to make this a bit more descriptive and also generic so `x_GPS` now `GNSSxECFPosition` and `x_LEO` now `SATxECFPosition`

Reply:
Let me see if Victoriya has any ideas on more descriptive names for these.

Reply:
@davidkuhl this is not ready yet... ;0)

will let you know when I generate a file, we will also bring in other Victoriya for certain to discuss convention